File: app/src/main/java/com/saint/struct/data/PagingDataSourceFactory.java
package com.saint.struct.data;

import androidx.annotation.NonNull;
import androidx.paging.DataSource;

import com.saint.struct.bean.WanListBean;

public class PagingDataSourceFactory extends DataSource.Factory<String, WanListBean> {

    @NonNull
    @Override
    public DataSource<String, WanListBean> create() {
        PagingOldDataSource dataSource = new PagingOldDataSource();
        return dataSource;
    }
}

File: app/src/main/java/com/saint/struct/data/PagingDataSourceFactory.kt
package com.saint.struct.data

import androidx.paging.DataSource
import com.saint.struct.bean.WanListBean

class PagingDataSourceFactory : DataSource.Factory<String, WanListBean>() {
    override fun create(): DataSource<String, WanListBean> {
        return PagingOldDataSource()
    }
}